components/eam/src/physics/cam/convect_deep.F90

```diff
@@ -1,428 +1,331 @@
-! Purpose: register fields with the physics buffer
-!----------------------------------------
-
-  
-  use physics_buffer, only : pbuf_add_field, dtype_r8
-  use zm_conv_intr, only: zm_conv_register
-  use phys_control, only: phys_getopts, use_gw_convect
-
-  implicit none
-
-  integer idx
-
-  ! get deep_scheme setting from phys_control
-  call phys_getopts(deep_scheme_out = deep_scheme)
-
-  select case ( deep_scheme )
-  case('ZM') !    Zhang-McFarlane (default)
-     call zm_conv_register
-  end select 
-
-! Add PBUF variables that are related to deep convection that 
-!  are expected by the E3SM code, no matter which deep convection scheme
-!  is used.
-
-! Flux of precipitation from deep convection (kg/m2/s)
-   call pbuf_add_field('DP_FLXPRC','global',dtype_r8,(/pcols,pverp/),dp_flxprc_idx) 
-
-! Flux of snow from deep convection (kg/m2/s) 
-   call pbuf_add_field('DP_FLXSNW','global',dtype_r8,(/pcols,pverp/),dp_flxsnw_idx) 
-
-! deep gbm cloud liquid water (kg/kg)
-   call pbuf_add_field('DP_CLDLIQ','global',dtype_r8,(/pcols,pver/), dp_cldliq_idx)  
-
-! deep gbm cloud liquid water (kg/kg)    
-   call pbuf_add_field('DP_CLDICE','global',dtype_r8,(/pcols,pver/), dp_cldice_idx) 
-  
-  call pbuf_add_field('ICWMRDP',    'physpkg',dtype_r8,(/pcols,pver/),icwmrdp_idx)
-  call pbuf_add_field('RPRDDP',     'physpkg',dtype_r8,(/pcols,pver/),rprddp_idx)
-  call pbuf_add_field('NEVAPR_DPCU','physpkg',dtype_r8,(/pcols,pver/),nevapr_dpcu_idx)
-  call pbuf_add_field('PREC_DP',    'physpkg',dtype_r8,(/pcols/),     prec_dp_idx)
-  call pbuf_add_field('SNOW_DP',   'physpkg',dtype_r8,(/pcols/),      snow_dp_idx)
-
-  call pbuf_add_field('LAMBDADPCU', 'physpkg', dtype_r8,(/pcols,pver/), lambdadpcu_idx)
-  call pbuf_add_field('MUDPCU',     'physpkg', dtype_r8,(/pcols,pver/), mudpcu_idx)
-  call pbuf_add_field('ICIMRDP',    'physpkg', dtype_r8,(/pcols,pver/), icimrdp_idx)
-
-  ! If WACCM gravity waves are on, output this field.
-  if (use_gw_convect) then
-     call pbuf_add_field('TTEND_DP','physpkg',dtype_r8,(/pcols,pver/),ttend_dp_idx)
-  end if
+!===================================================================================================
+
+subroutine convect_deep_register()
+   !----------------------------------------------------------------------------
+   ! Purpose: register fields with the physics buffer
+   !----------------------------------------------------------------------------
+   use phys_control,   only: phys_getopts, use_gw_convect
+   use physics_buffer, only: pbuf_add_field, dtype_r8
+   use zm_conv_intr,   only: zm_conv_register
+   !----------------------------------------------------------------------------
+   integer idx
+
+   ! get deep_scheme setting from phys_control
+   call phys_getopts(deep_scheme_out = deep_scheme)
+
+   select case ( deep_scheme )
+   case('ZM') ! Zhang-McFarlane
+      call zm_conv_register
+   end select 
+
+   ! Add PBUF variables related to deep convection that
```

Comment:
Comment is cutoff here. I think the following couple lines were meant to be reformatted but not removed.

Reply:
Good catch, I meant to delete "that"

Reply:
fixed.